classes/item.rb

```diff
@@ -0,0 +1,28 @@
+require 'date'
+
+class Item
+  attr_accessor :author, :genre, :source, :label, :publish_date
+  attr_reader :id, :archived
+
+  @@id = 0
+
+  def initialize(author: 0, genre: 0, source: 0, label: 0, archived: false, date: DateTime.now.to_s.slice(0, 10))
+    @id = @@id
+    @@id += 1
+    @author = author
+    @genre = genre
+    @source = source
+    @label = label
+    @archived = archived
+    @publish_date = date
+  end
```

Comment:
I think the way you used the id is not a best practice, you can use instead 
@id = Rand(1...100)
also, we don't need resources since we are 3 members in the group.

Reply:
@mwafrika why its not best practice? rand can give same number more than one time. class variable never gives out same number.

Reply:
I know we dont need `source`. I did it to be in line with the diagram. Its keyword argument so it won't affect how we call the constructor.